packages/text/src/Text.ts

```diff
@@ -509,55 +505,68 @@ export class Text extends Sprite
         if (style.fillGradientType === TEXT_GRADIENT.LINEAR_VERTICAL)
         {
             // start the gradient at the top center of the canvas, and end at the bottom middle of the canvas
-            gradient = this.context.createLinearGradient(width / 2, padding, width / 2, height);
+            gradient = this.context.createLinearGradient(width / 2, padding, width / 2, height + padding);
 
             // we need to repeat the gradient so that each individual line of text has the same vertical gradient effect
             // ['#FF0000', '#00FF00', '#0000FF'] over 2 lines would create stops at 0.125, 0.25, 0.375, 0.625, 0.75, 0.875
-            totalIterations = (fill.length + 1) * lines.length;
-            currentIteration = 0;
 
             // There's potential for floating point precision issues at the seams between gradient repeats.
             // The loop below generates the stops in order, so track the last generated one to prevent
             // floating point precision from making us go the teeniest bit backwards, resulting in
             // the first and last colors getting swapped.
             let lastIterationStop = 0;
 
+            const metrics = TextMetrics.measureText(this.text, style, style.wordWrap, this.canvas);
```

Comment:
You already have a measurement of the text outside of this function from line 183. So pass this in as a 3rd param to this function on line 247
`context.fillStyle = this._generateFillStyle(style, lines, measured);`
and you've saved a call to measureText

Reply:
Good idea, no point measuring twice if we can reuse one.